pkg/frontend/http/extension/server.go

```diff
@@ -0,0 +1,241 @@
+// Copyright 2023 The Kelemetry Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package extension
+
+import (
+	"context"
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/gin-gonic/gin"
+	"github.com/jaegertracing/jaeger/model"
+	uiconv "github.com/jaegertracing/jaeger/model/converter/json"
+	"github.com/jaegertracing/jaeger/storage/spanstore"
+	"github.com/sirupsen/logrus"
+	"github.com/spf13/pflag"
+	"k8s.io/utils/clock"
+
+	"github.com/kubewharf/kelemetry/pkg/frontend/clusterlist"
+	jaegerreader "github.com/kubewharf/kelemetry/pkg/frontend/reader"
+	tfconfig "github.com/kubewharf/kelemetry/pkg/frontend/tf/config"
+	pkghttp "github.com/kubewharf/kelemetry/pkg/http"
+	"github.com/kubewharf/kelemetry/pkg/manager"
+	"github.com/kubewharf/kelemetry/pkg/metrics"
+	"github.com/kubewharf/kelemetry/pkg/util/shutdown"
+)
+
+func init() {
+	manager.Global.Provide("jaeger-extension-server", NewRedirectServer)
+}
+
+type options struct {
+	enable bool
+}
+
+func (options *options) Setup(fs *pflag.FlagSet) {
+	fs.BoolVar(&options.enable, "jaeger-extension-server-enable", false, "enable extension server for frontend")
```

Comment:
What about calling this "trace AP" instead? Then we call everything (including this one and the redirect endpoint) the "extension server"

Reply:
good idea, updated